pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -231,16 +233,25 @@ func (c *Reconciler) reconcileConfiguration(ctx context.Context, service *v1alph
 	// TODO(#642): Remove this (needed to avoid continuous updates)
 	desiredConfig.Spec.Generation = config.Spec.Generation
 
-	if equality.Semantic.DeepEqual(desiredConfig.Spec, config.Spec) {
-		// No differences to reconcile.
+	diff := false
+	if !equality.Semantic.DeepEqual(desiredConfig.Spec, config.Spec) {
+		diff = true
+		logger.Infof("Reconciling configuration spec diff (-desired, +observed): %v", cmp.Diff(desiredConfig.Spec, config.Spec))
+	}
+	if !config.IsInheritedAnnotationsEqual(desiredConfig.GetObjectMeta()) {
+		diff = true
+		logger.Infof("Reconciling configuration inherited annotation diff (-desired, +observed): %v", cmp.Diff(desiredConfig.Annotations, config.Annotations))
+	}
+	if diff == false {
+		logger.Debugf("No configuration differences to reconcile.")
 		return config, nil
 	}
-	logger.Infof("Reconciling configuration diff (-desired, +observed): %v", cmp.Diff(desiredConfig.Spec, config.Spec))
 
 	// Don't modify the informers copy
 	existing := config.DeepCopy()
 	// Preserve the rest of the object (e.g. ObjectMeta)
 	existing.Spec = desiredConfig.Spec
+	existing.InheritAnnotations(desiredConfig.GetObjectMeta())
```

Comment:
This CL overall ended up looking very little like what I had in mind. :)

I would have expected us to copy the annotations from Revision -> KPA in this style: https://github.com/knative/serving/blob/516c6a6aec1e23967ab61c356c518730db1b101f/pkg/reconciler/v1alpha1/service/resources/configuration.go#L38

As for the Service -> Configuration -> Revision inheritance, if anything I would have expected something closer to the above, but my inclination for this PR is to say: "Service/Revision embed a RevisionTemplateSpec for the precise reason that you can set metadata on Revision, so this annotation should simply be put there directly."

e.g.
```yaml
spec:
  revisionTemplate:
    metadata:
      annotations:
        autoscaling.knative.dev/class: kpa
    spec:
      container:
        image: busybox
```

My motivation for this bias vs. inheritance is that the former results in Generation bumps on updates and the latter does not.  In other words, if I change the annotation on the outer thing, I don't get a new Revision right away, but if I change the annotation on the inner thing, I do.

Happy to talk thru this f2f tomorrow, or VC Friday if it would be helpful.

Reply:
Summary of conversation with @mattmoor 
- We need to think more about how to do inheritance in general.  Table it for now.
- We also need to think more about Namespace inheritance.
- Set autoscaling annotations on the Configuration.Spec.RevisionTemplate so it stamps out a new Revision.

This unblocks the core functionality of this extension and we can come back to Service and Namespace inheritance.

Reply:
thanks, @josephburnett!